test/registration_spec.rb

```diff
@@ -201,38 +201,48 @@
 
   describe "#get_updates_list" do
     let(:base_product) { { "name" => "base" } }
+    let(:installer_update_base_product) { { "name" => "self_update_id" } }
```

Comment:
this should use variable and not string?

Reply:
when it is returned means that the self_update_id variable is not empty if it is empty then the method should return nil not a hash

Reply:
I really do not like this as I do not know if it is coincident name string or not. is posible to use here `"name" => self_update_id`? as it looks like connected values for me. And if it is not connect, please use different name.

Reply:
As I wrote below, using a real value like `"SLES"` here would make it clear.